src/accelerate/data_loader.py

```diff
@@ -870,6 +870,10 @@ def prepare_data_loader(
         sampler = getattr(dataloader.sampler, "sampler", None)
     else:
         sampler = getattr(dataloader.batch_sampler, "sampler", None)
+    if isinstance(dataloader.sampler, RandomSampler) and state.distributed_type == DistributedType.XLA:
+        seed=torch.Generator().manual_seed(42)
```

Comment:
nit with the naming: `manual_seed` doesn't return the seed, it returns a _seeded generator_: https://critique.corp.google.com/cl/577336396

So `seed` should instead be `seeded_generator`, `rng`, etc (whichever name matches the existing style the best).